puppet/modules/web/manifests/vhost/stagingyum.pp

```diff
@@ -0,0 +1,53 @@
+# @summary Set up the yum vhost
+# @api private
+class web::vhost::stagingyum (
+  Stdlib::Fqdn $servername = 'stagingyum.theforeman.org',
+  Stdlib::Absolutepath $yum_directory = '/var/www/vhosts/stagingyum/htdocs',
+  String $user = 'yumrepostage',
+) {
+  $yum_directory_config = [
+    {
+      path            => $yum_directory,
+      options         => ['Indexes', 'FollowSymLinks', 'MultiViews'],
+      expires_active  => 'on',
+      expires_default => 'access plus 2 minutes',
+    },
+    {
+      path            => '.+\.(bz2|gz|rpm|xz)$',
+      provider        => 'filesmatch',
+      expires_active  => 'on',
+      expires_default => 'access plus 30 days',
+    },
+    {
+      path            => 'repomd.xml',
+      provider        => 'files',
+      expires_active  => 'on',
+      expires_default => 'access plus 2 minutes',
+    },
+  ]
+
+  secure_ssh::receiver_setup { $user:
+    user           => $user,
+    foreman_search => 'host ~ node*.jenkins.osuosl.theforeman.org and (name = external_ip4 or name = external_ip6)',
+    script_content => template('web/deploy-stagingyum.sh.erb'),
+  }
+
+  web::vhost { 'stagingyum':
+    servername    => $servername,
+    docroot       => $yum_directory,
+    docroot_owner => $user,
+    docroot_group => $user,
+    docroot_mode  => '0755',
+    directories   => $yum_directory_config,
+  }
+
+  ['HEADER.html', 'robots.txt'].each |$filename| {
+    file { "${yum_directory}/${filename}":
+      ensure  => file,
+      owner   => $user,
+      group   => $user,
```

Comment:
I wouldn't mind making these owned by root so rsync can't overwrite them, even accidentally. Otherwise it may flip back & forth.